maven/bnd-testing-maven-plugin/src/main/java/aQute/bnd/maven/testing/plugin/TestingMojo.java

```diff
@@ -103,94 +93,11 @@ private void testing(File runFile) throws MojoExecutionException, Exception, IOE
 			}
 
 			if (resolve) {
-				resolve(run);
+				run.resolve(failOnChanges);
```

Comment:
So for this PR, I think you need to change the new Bndrun.resolve method such that you can either mutate the source bndrun file or not. And the method should probably return the new runbundles value, so that caller can use it if they don't request the mutation.

Then we can follow with a new PR to add the new bnd-resolve-maven-plugin.